pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go

```diff
@@ -125,14 +125,21 @@ func getAdditionalHosts(domainList *[]string, serviceHost string, config *v1beta
 		// len(subdomain) != 0 means we have found the subdomain.
 		// If the list of the additionalIngressDomains is not empty, we will append the valid host created by the
 		// additional ingress domain.
+		// Deduplicate the domains in the additionalIngressDomains, making sure that the returned additionalHosts
+		// do not have duplicate domains.
+		deduplicateMap := map[string]bool{}
 		for _, domain := range *config.AdditionalIngressDomains {
-			host := fmt.Sprintf("%s%s", subdomain, domain)
-			if err := validation.IsDNS1123Subdomain(host); len(err) > 0 {
-				log.Error(fmt.Errorf("The domain name %s in the additionalIngressDomains is not valid", domain),
-					"Failed to get the valid host name")
-				continue
+			// If the domain is redundant, go to the next element.
+			if !deduplicateMap[domain] {
```

Comment:
just to confirm for boolean map this is good enough, we do not need to check the existence flag.

Reply:
Yes, this is good use of the boolean map.